core/platform/image/gae_image_services_test.py

```diff
@@ -0,0 +1,105 @@
+# coding: utf-8
+#
+# Copyright 2018 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for methods in the gae_image_services."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+import os
+
+from constants import constants
+from core.platform.image import gae_image_services
+from core.tests import test_utils
+import feconf
+import python_utils
+
+
+class GaeImageServicesUnitTests(test_utils.GenericTestBase):
+    """Tests for gae_image_services."""
+
+    TEST_IMAGE_WIDTH = 3000
+    TEST_IMAGE_HEIGHT = 2092
+
+    def setUp(self):
+        super(GaeImageServicesUnitTests, self).setUp()
+        with python_utils.open_file(
+            os.path.join(feconf.TESTS_DATA_DIR, 'dummyLargeImage.jpg'),
+            mode='rb', encoding=None) as f:
+            self.raw_image = f.read()
+
+    def test_get_image_dimensions(self):
+        compressed_image = (
+            gae_image_services.compress_image(self.raw_image, 0.5))
+        height, width = (
+            gae_image_services.get_image_dimensions(compressed_image))
+        # The google app engine images api sometimes rounds off by 1 pixel
```

Comment:
These tests do not need to refer at all to the GAE images API. From now on we are only using the pillow implementation.

Ditto below.

(Also maybe this means you can do an exact comparison?)

Reply:
Done